baml_language/crates/baml_hir/src/lib.rs

```diff
@@ -687,16 +741,35 @@ pub fn lower_type_ref(node: &baml_syntax::ast::TypeExpr) -> TypeRef {
         "video" => TypeRef::Video,
         "pdf" => TypeRef::Pdf,
         _ => {
+            // Check for string literal types
+            if (text.starts_with('"') && text.ends_with('"'))
+                || (text.starts_with('\'') && text.ends_with('\''))
+            {
+                // String literal type; extract the value
+                let inner = &text[1..text.len() - 1];
+                return TypeRef::StringLiteral(inner.to_string());
+            }
```

Comment:
This seems to be an early return that cuts recursion for string literals, I think it will not handle the case where string literals are optional:
`"literal"?`